src/desktop/apps/partner2/client/shows_grid.coffee

```diff
@@ -109,17 +109,16 @@ module.exports = class PartnerShowsGridView extends Backbone.View
           @remainingUpcoming = @remainingUpcoming.concat moreShows.upcoming().models
           @remainingPast = @remainingPast.concat moreShows.past().models
           shows = @getRemainingShows type #get updated shows list from fetch
-          displayMore = shows.length > 30
     else
-      displayMore = false
-
-    $(".#{type} .loading-spinner").remove()
-    $(".#{type} .partner2-shows-container").append showFiguresTemplate
-      shows: shows
-      type: type
-      displayMore: displayMore
-      isFeatured: false
-    @sliceRemaining type
+      Promise.resolve()
+
+    getMoreShows.then =>
+      $(".#{type} .loading-spinner").remove()
+      $(".#{type} .partner2-shows-container").append showFiguresTemplate
+        shows: shows
+        type: type
+        isFeatured: false
+      @sliceRemaining type
```

Comment:
Previously this would perform the fetch but fail to wait for it before rendering.